node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/LoadBalancerProvisioner.java

```diff
@@ -60,8 +60,11 @@ public LoadBalancerProvisioner(NodeRepository nodeRepository, LoadBalancerServic
         // Read and write all load balancers to make sure they are stored in the latest version of the serialization format
         for (var id : db.readLoadBalancerIds()) {
             try (var lock = db.lock(id.application())) {
-                var loadBalancer = db.readLoadBalancer(id);
-                loadBalancer.ifPresent(db::writeLoadBalancer);
+                // TODO(mpolden): Remove inner lock
+                try (var innerLock = db.configLock(id.application())) {
```

Comment:
This code is executed on bootstrap, so if a big deployment is happening concurrently the above db.lock() will time out.  But bootstrap redeployments are also vulnerable to this.

Reply:
Yes. This is also why I left out locking in the `NodeRepository` constructor even though it's wrong. If a lock timeout happens at construction time it should recover though as the container will restart and try to construct the components again, and by then deployments are hopefully done.